TESTS/netsocket/tls/tlssocket_send_unconnected.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2019, ARM Limited, All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "mbed.h"
+#include "TLSSocket.h"
+#include "greentea-client/test_env.h"
+#include "unity/unity.h"
+#include "utest.h"
+#include "tls_tests.h"
+
+using namespace utest::v1;
+
+void TLSSOCKET_SEND_UNCONNECTED()
+{
+    TLSSocket sock;
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.open(get_interface()));
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.set_root_ca_cert(tls_global::cert));
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_NO_CONNECTION, sock.send("12345", 5));
+    TEST_ASSERT_EQUAL(NSAPI_ERROR_OK, sock.close());
```

Comment:
Should NSAPI_ERROR_NO_CONNECTION be also here the only accepted return code?

Reply:
Hmm.. Actually, you can call `Socket::close()` before connecting it.

Close is originally opposite of `Socket::open()` but in fact it serves as disconnect call as well for TCPSocket.

Confusion partially comes from POSIX where `socket()` creates a socket and `close()` does both, disconnection and destruction of the file handle.